e2e/specs/st_multiselect.spec.js

```diff
@@ -240,9 +250,41 @@ describe("st.multiselect", () => {
             "value 6: []" +
             "value 7: []" +
             "value 8: []" +
-            "value 9: ['male']" +
+            "value 9: []" +
+            "value 10: ['male']" +
             "multiselect changed: True"
         );
     });
   });
+
+  describe("when using max_selections for st.multiselect", () => {
+    it("should show the correct text when maxSelections is reached", () => {
+      selectNthOptionForKthMultiselect(0, 8)
+      cy.getIndexed(".stMultiSelect", 8).then(el => {
+        cy
+          .wrap(el)
+          // recheck that we have hit the limit
+          .find("input")
+          .click()
+          .get("li")
+          .first()
+          .should("have.text", "You can only select up to 1 options. Remove an option first.")
```

Comment:
nit (technically for `Multiselect.tsx` and not here): Hm, we probably want to special-case the scenario where there's only 1 option so the grammar is correct here